core/src/main/java/org/sonarsource/sonarlint/core/event/ConnectionCredentialsChangedEvent.java

```diff
@@ -19,11 +19,14 @@
  */
 package org.sonarsource.sonarlint.core.event;
 
-public class ConnectionCredentialsChangedEvent {
+import org.springframework.context.ApplicationEvent;
+
+public class ConnectionCredentialsChangedEvent extends ApplicationEvent {
 
   private final String connectionId;
 
-  public ConnectionCredentialsChangedEvent(String connectionId) {
+  public ConnectionCredentialsChangedEvent(Object source, String connectionId) {
+    super(source);
```

Comment:
Do you know what the purpose of this `source` is? It is a bit painful to pass an extra object for each event

Reply:
In fact, it is now possible with recent version of Spring to not inherit from the framework abstract class. I will change.